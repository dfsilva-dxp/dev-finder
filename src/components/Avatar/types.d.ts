export interface IAvatarProps {
  url_img?: string;
}
