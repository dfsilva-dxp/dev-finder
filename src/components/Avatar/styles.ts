import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    border: 2px solid ${theme.colors.light};

    width: 120px;
    height: 120px;
    border-radius: 50%;
    overflow: hidden;
    padding: 4px;

    img {
      width: 100%;
      border-radius: 50%;
    }
  `}
`;
