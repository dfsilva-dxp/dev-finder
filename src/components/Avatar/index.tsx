import { Container } from "./styles";
import { IAvatarProps } from "./types";

export default function Avatar({ url_img }: IAvatarProps) {
  return (
    <Container>
      <img src={url_img} alt="User Avatar" title="User Avatar" />
    </Container>
  );
}
