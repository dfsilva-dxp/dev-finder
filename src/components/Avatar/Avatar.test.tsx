import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/test/helpers";

import Avatar from ".";

describe("<Avatar/>", () => {
  it("should render currectly", () => {
    const { container } = renderWithTheme(<Avatar url_img="my_url" />);

    expect(screen.getByRole("img", { name: /user avatar/i })).toHaveAttribute(
      "src",
      "my_url"
    );

    expect(
      screen.getByRole("img", { name: /user avatar/i }).parentElement
    ).toHaveStyle({
      width: "120px",
      height: "120px",
      "border-radius": "50%",
      overflow: "hidden",
      padding: "4px"
    });

    expect(container.parentElement).toMatchSnapshot();
  });
});
