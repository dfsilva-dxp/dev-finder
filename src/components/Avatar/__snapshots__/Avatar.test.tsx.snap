// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar/> should render currectly 1`] = `
.c0 {
  border: 2px solid #1F2A48;
  width: 120px;
  height: 120px;
  border-radius: 50%;
  overflow: hidden;
  padding: 4px;
}

.c0 img {
  width: 100%;
  border-radius: 50%;
}

<body>
  <div>
    <div
      class="c0"
    >
      <img
        alt="User Avatar"
        src="my_url"
        title="User Avatar"
      />
    </div>
  </div>
</body>
`;
